Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -153,32 +151,79 @@ public function deleteCookie(string $cookieName): void
      * @param string $parameterName
      * @param mixed $value
      * @return void
-     * @api
+     * @deprecated since Flow 7.0 use setHttpHeader or replaceHttpResponse instead. See also #linkToDocs
      */
     public function setComponentParameter(string $componentClassName, string $parameterName, $value): void
     {
-        if (!isset($this->componentParameters[$componentClassName])) {
-            $this->componentParameters[$componentClassName] = [];
+        if ($componentClassName === SetHeaderComponent::class) {
+            $this->setHttpHeader($parameterName, $value);
+            return;
         }
-        $this->componentParameters[$componentClassName][$parameterName] = $value;
+
+        if ($componentClassName === ReplaceHttpResponseComponent::class && $parameterName === 'response') {
+            $this->replaceHttpResponse($value);
+            return;
+        }
+        throw new \InvalidArgumentException('todo');
     }
 
     /**
-     * @return string
+     * Set the specified header in the response, overwriting any previous value set for this header.
+     *
+     * @param string $headerName The name of the header to set
+     * @param array|string|\DateTime $headerValue An array of values or a single value for the specified header field
+     * @return void
      */
-    public function getContent(): string
+    public function setHttpHeader(string $headerName, $headerValue): void
     {
-        $content = $this->content->getContents();
-        $this->content->rewind();
-        return $content;
+        if ($headerValue instanceof \DateTimeInterface) {
```

Comment:
This is taken from the `Headers` class, which should eventually replace this implementation and add more response API methods.

Reply:
We should add this as comment IMO

Reply:
```suggestion
        // This is taken from the Headers class, which should eventually replace this implementation and add more response API methods.
        if ($headerValue instanceof \DateTimeInterface) {
```